s3/replication/common/src/s3replicationcommon/s3_head_object.py

```diff
@@ -42,9 +42,357 @@ def __init__(self, session, request_id,
+        return self._resp_expires
+
+    def get_last_modified(self):
+        """Get last creation date of object.
+
+        Returns
+        -------
+            str: date of the object.
+        """
+        self._resp_last_modified = self._response_headers.get(
+            "Last-Modified", None)
+        return self._resp_last_modified
+
+    def get_server_name(self):
+        """Get server name.
+
+        Returns
+        -------
+            str: server name (SeagateS3 / AmazonS3).
+        """
+        self._resp_server_name = self._response_headers.get(
+            "Server", None)
+        return self._resp_server_name
+
+    def get_x_amz_archive_status(self):
+        """Get archive state of the object.
+
+        Returns
+        -------
+            str: archieve state (ARCHIVE_ACCESS / DEEP_ARCHIVE_ACCESS)
+        """
+        self._resp_archive_status = self._response_headers.get(
+            "x-amz-archive-status", None)
+        return self._resp_archive_status
+
+    def get_x_amz_delete_marker(self):
+        """Get delete marker status for object.
+
+        Returns
+        -------
+            bool: True if object retrived was a Delete Marker, else False.
+        """
+        self._resp_delete_marker = self._response_headers.get(
+            "x-amz-delete-marker", None)
+        if self._resp_delete_marker is not None:
+            self._resp_delete_marker = bool(self._resp_delete_marker)
+        return self._resp_delete_marker
+
+    def get_x_amz_expiration(self):
+        """Get expiration configuration of object.
+
+        Returns
+        -------
+            str: expiry date and rule-id, if enabled.
+        """
+        self._resp_expiration = self._response_headers.get(
+            "x-amz-expiration", None)
+        return self._resp_expiration
+
+    def get_x_amz_missing_meta(self):
+        """Get missing metadata entries of object.
+
+        Returns
+        -------
+            int: value of the number of unprintable metadata entries.
+        """
+        self._resp_missing_data = self._response_headers.get(
+            "x-amz-missing-meta", None)
+        if self._resp_missing_data is not None:
+            self._resp_missing_data = int(self._resp_missing_data)
+        return self._resp_missing_data
+
+    def get_x_amz_mp_parts_count(self):
+        """Get part counts of object.
+
+        Returns
+        -------
+            int: total part count of an object.
+        """
+        self._resp_parts_count = self._response_headers.get(
+            "x-amz-mp-parts-count", None)
+        if self._resp_parts_count is not None:
+            self._resp_parts_count = int(self._resp_parts_count)
+        return self._resp_parts_count
+
+    def get_x_amz_object_lock_legal_hold(self):
+        """Get legal hold status value for the object.
+
+        Returns
+        -------
+            str: ON if a legal hold is in effect for the object, else OFF.
```

Comment:
will it be capital response? or you have put it like this to emphasize?

Reply:
yes, it will be a capital response, having return value either ON/OFF.